pkg/document/proxy/text_proxy.go

```diff
@@ -70,3 +70,30 @@ func (p *TextProxy) Edit(from, to int, content string) *TextProxy {
 	}
 	return p
 }
+
+// Select stores that the given range has been selected.
+func (p *TextProxy) Select(from, to int) *TextProxy {
```

Comment:
It seems to be outside the scope of the current PR title and content. I think we can revise the title and content so we can see that this feature has been added to this PR.
Let me know what you think.

Reply:
Thank you for the review. I have added the missing features to increase coverage. I'll add it to the body of this commit.